Tests/Unit/Controller/SitemapControllerTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * This file is part of the PrestaSitemapBundle package.
+ *
+ * (c) PrestaConcept <www.prestaconcept.net>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Presta\SitemapBundle\Tests\Unit\Controller;
+
+use PHPUnit\Framework\TestCase;
+use Presta\SitemapBundle\Controller\SitemapController;
+use Presta\SitemapBundle\Service\GeneratorInterface;
+use Presta\SitemapBundle\Sitemap\Sitemapindex;
+use Presta\SitemapBundle\Sitemap\Urlset;
+use Prophecy\Prophecy\ObjectProphecy;
+use Symfony\Component\HttpFoundation\Response;
+
+class SitemapControllerTest extends TestCase
+{
+    private const TTL = 3600;
+
+    /**
+     * @var GeneratorInterface|ObjectProphecy
+     */
+    private $generator;
+
+    public function setUp(): void
+    {
+        $this->generator = $this->prophesize(GeneratorInterface::class);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->generator = null;
```

Comment:
This looks odd, such cleanup should be done by phpunit automatically.

Reply:
I'm not sure phpunit is cleaning anything. And I saw that kind of things in many different places, like in Symfony : https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php#L45
This is something I'm doing in every tests I write, because I like to be in control of the "state" of tests : isolation is the key.

Reply:
Those are static vars though

Reply:
You're right ! 
I left a debug in a test `__construct` method, looks like phpunit is constructing it for each test method in that test case... Will save me a lot of work ^^